### PR TITLE
Changed generated python client to accept None as request body

### PR DESCRIPTION
**Pull request by telday**
This allows optional request bodies to work properly, previously they would always fail when body was None.

Resolves: #8300

Signed-off-by: Ellis Wright <ellis.wright@cyberark.com>

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @Jyhess (2019/01) @arun-nalla (2019/11) @spacether (2019/11)

### CODE REVIEW

**Comment by spacether** [comment]
Hey there. So I am concerned that we are not coving all of our use cases here.
- Required can be True or False
- Body schema can be nullable or not nullable
So sending None/null as a body is valid and should be different than omitting the body parameter.
How do you feel about reworking our code to cover these two use cases?
1) send + receive None nullable body (required= True or False, nullable=True)
2) send and receive request with body omitted (required=False)

Some ways to do this would be to:
- extract body from kwargs in the method `request`
- set the body default value to a unset_value class parameter:
```
class RESTClientObject:
    unset_value = object()
    def request(self, method, url, query_params=None, headers=None,
        body=unset_value, post_params=None, _preload_content=True,
        _request_timeout=None):
```

**Comment by telday** [comment]
@spacether Just added another chunk of code which checks if the body is nullable then converts the value to an empty string if it is not. I struggled with where to put this logic, let me know what you think.

File: modules/openapi-generator/src/main/resources/python/api.mustache [hunk 1]
```diff
@@ -123,6 +123,12 @@ class {{classname}}(object):
             kwargs['{{paramName}}'] = \
                 {{paramName}}
 {{/requiredParams}}
+
+            if 'body' not in kwargs:
+                body = 'body' in self.params_map['all']
+                body_nullable = 'body' in self.params_map['nullable']
+                if body and not body_nullable:
+                    kwargs['body'] = ''
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python/api.mustache:131]: Why are we setting body to empty string here?

**Comment by spacether** [comment]
@telday thank you for making those changes. It's not clear to me if this code will code the use cases that I asked for.
Can you add tests showing it working?
The cases that will prove that this is working are:
- required body parameter, body nullable, show transmit and receiving none working
- optional body parameter, body nullable, show transmit and receiving none working

One can edit the sample spec file to add those endpoints here:
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml
and the apple schema is nullable and could be used for these endpoints.

For the tests, one could add them to the tag fake_api and add a test like we have here:
https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python/tests_manual/test_fake_api.py#L212

**Pull request closed without merging**